sources/api/netdog/src/cli/write_primary_interface_status.rs

```diff
@@ -0,0 +1,116 @@
+use super::{error, primary_interface_name, Result};
+use crate::dns::DnsSettings;
+use crate::networkd_status::NetworkDInterfaceStatus;
+use crate::{CURRENT_IP, PRIMARY_SYSCTL_CONF, SYSCTL_MARKER_FILE, SYSTEMD_SYSCTL};
+use argh::FromArgs;
+use snafu::{ensure, ResultExt};
+use std::fmt::Write;
+use std::fs;
+use std::net::IpAddr;
+use std::path::Path;
+use std::process::Command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "write-primary-interface-status")]
+/// Updates the various files needed when responding to events
+pub(crate) struct WritePrimaryInterfaceStatusArgs {}
+
+/// Updates the various files needed when responding to events
+pub(crate) fn run() -> Result<()> {
+    let primary_interface = primary_interface_name()?;
+
+    let primary_link_status = NetworkDInterfaceStatus::new(primary_interface.clone())
+        .context(error::NetworkDInterfaceStatusSnafu {})?;
+
+    // Write out current IP
+    let primary_ip = &primary_link_status
+        .primary_address()
+        .context(error::PrimaryInterfaceAddressSnafu {})?;
+    write_current_ip(primary_ip)?;
+
+    // Write out resolv.conf
+    write_resolv_conf(&primary_link_status)?;
+
+    // If we haven't already, set and apply default sysctls for the primary network
+    // interface
+    if !Path::exists(Path::new(PRIMARY_SYSCTL_CONF)) {
+        write_interface_sysctl(primary_interface, PRIMARY_SYSCTL_CONF)?;
+    };
+
+    // Execute `systemd-sysctl` with our configuration file to set the sysctls
+    if !Path::exists(Path::new(SYSCTL_MARKER_FILE)) {
+        let systemd_sysctl_result = Command::new(SYSTEMD_SYSCTL)
+            .arg(PRIMARY_SYSCTL_CONF)
+            .output()
+            .context(error::SystemdSysctlExecutionSnafu)?;
+        ensure!(
+            systemd_sysctl_result.status.success(),
+            error::FailedSystemdSysctlSnafu {
+                stderr: String::from_utf8_lossy(&systemd_sysctl_result.stderr)
+            }
+        );
+
+        fs::write(SYSCTL_MARKER_FILE, "").unwrap_or_else(|e| {
+            eprintln!(
+                "Failed to create marker file {}, netdog may attempt to set sysctls again: {}",
+                SYSCTL_MARKER_FILE, e
+            )
+        });
+    }
+    Ok(())
+}
+
+/// Write the default sysctls for a given interface to a given path
+fn write_interface_sysctl<S, P>(interface: S, path: P) -> Result<()>
+where
+    S: AsRef<str>,
+    P: AsRef<Path>,
+{
+    let interface = interface.as_ref();
+    let path = path.as_ref();
+    // TODO if we accumulate more of these we should have a better way to create than format!()
+    // Note: The dash (-) preceding the "net..." variable assignment below is important; it
+    // ensures failure to set the variable for any reason will be logged, but not cause the sysctl
+    // service to fail
```

Comment:
I don't love copy/pasting a TODO from another source file that says to make a better way. Either we should get rid of this comment if we're not actually going to do it, or we should do it. It reminds me of when we had "TODO: stop hard-coding `/run/api.sock`" in a dozen different places.

Reply:
This is reasonable. I think we have found that we are not actually accumulating these so its probably fine to remove the TODO. The not about the Dash seems useful though so we probably should keep this.